### DISCUSSION TITLE
System Requirements

### DISCUSSION THREAD

**Discussion by marcusheBG** [IBM/mac-ibm-notifications#127] [Q&A] [answered]
Hello! What minimum macOS version is need for IBMNotifier - I believe it is 10.15 Catalina?

We have some older Macs on 10.14 Mojave we would like to use IBMNotifier for to have a process to get them upgraded to current macOS.

Could Xcode build a version for 10.14 without much modification? Only problem is then we need to sign with a developer ID,

Thank you!!

**Comment by SMartorelli** [comment, marked as answer]
The builded version of IBM Notifier supports macOS 10.15+ it is actually possible to build IBM Notifier to be compatible with 10.14 but we don't provide it directly because we don't QA 10.14.
In order to do so you need to change all the target deployment target to 10.14

<img width="1096" alt="Screenshot 2022-09-20 at 08 34 15" src="https://user-images.githubusercontent.com/23209655/191184695-cc95b95b-bde2-404e-bf4a-67c4c174d978.png">

And then comment the `allowPictureInPicturePlayback` line in `NotificationAgent/Shared/Views/AccessoryViews/VideoAccessoryView.swift` file

<img width="432" alt="Screenshot 2022-09-20 at 08 34 01" src="https://user-images.githubusercontent.com/23209655/191184920-bb216bc2-8e6a-48cf-aad3-7ebca7769baa.png">

This should allow you to build, archive and sing the tool for macOS 10.14.

**Reply by marcusheBG** [reply]
Thank you so much!!

We actually did just that - and the build completed - however when running the custom compiled build as described on 10.14 Mojave - the app icon shows the circle / slash and double-clicking the app states that "10.14 is required" even though it is running on 10.14.

Any ideas? Thank you again!!

**Reply by SMartorelli** [reply]
That's definitely strange. On which Mojave version you are?

**Reply by marcusheBG** [reply]
Thank you so much! We are on 10.14.6. Here is a screenshot of the error. Any ideas would be much appreciated. 
![IBMNotifier10 14 6CustomCompiledError](https://user-images.githubusercontent.com/90919338/191348117-ef0b10af-8951-4c1b-b0b3-d82ac9d090fe.png)

**Reply by marcusheBG** [reply]
<img width="1441" alt="Screen Shot 2022-09-20 at 3 35 57 PM" src="https://user-images.githubusercontent.com/90919338/191348914-d867b436-7221-4cdd-976a-b63442121cd3.png">
<img width="1410" alt="Screen Shot 2022-09-20 at 3 36 03 PM" src="https://user-images.githubusercontent.com/90919338/191348924-15515292-e13f-424a-9b98-a8ab2a8b23a3.png">

**Reply by SMartorelli** [reply]
Have you changed the deployment target for all the 5 component targets? No need for the test targets

**Reply by marcusheBG** [reply]
Yes thank you - we have changed the deployment target for all five components!

**Reply by SMartorelli** [reply]
And still not working right?

**Reply by marcusheBG** [reply]
Correct - all five deployment targets (including the test ones) have been changed to 10.14 target and it is still not working! Thank you.

**Reply by SMartorelli** [reply]
Well try to clean the build folder (Product -> Clean Build Folder) and reset packages cache (File -> Packages -> Reset Packages Cache) and then archive it again. I'm out of option other than this because the project doesn't use any API that isn't available with 10.14 (except the PiP stuff).

**Reply by marcusheBG** [reply]
<img width="377" alt="Screen Shot 2022-09-22 at 12 21 26 PM" src="https://user-images.githubusercontent.com/90919338/191800435-d7d9e2a5-8932-4da6-99f9-b28bf8fae96c.png">
Thank you so much. We just did that and same result. However we realized it seems the App is only being built for Apple Silicon as admins are using M1 Macs. How can we be sure to set it to build as a Universal Binary for Intel while on an M1 Mac? Thank you!

**Reply by SMartorelli** [reply]
When archived the app is built as Universal Binary. If you're debugging it on Xcode its built using just the current machine setup so the local build will produce an Apple Silicon app on Apple Silicon devices.

**Reply by marcusheBG** [reply]
Thank you SO MUCH @SMartorelli! That did the trick! Thank you for supporting our rudimentary questions! We will now have a workflow to enforce macOS upgrades via pop-up for these older 10.14 Macs.
